backend/apps/ifc_validation/checks/header_policy/validate_header.py

```diff
@@ -128,19 +165,69 @@ def populate_header(cls, values):
+        """
+        header_description_text = ' '.join(v)
         try:
-            if not v in list(map(lambda desc: desc.replace(" ", ""), allowed_descriptions.get(schema_identifier))):
-                values.data.get('validation_errors').append(values.field_name)
-            if not v: # description field is mandatory
-                values.data.get('validation_errors').append(values.field_name)
+            parsed_description = parse_mvd(header_description_text)
         except:
-            pass  ## ---> what to do with invalid schema identifier? (IFC4X3)
-        return (v,)
+            pass
+        view_definitions = parsed_description.mvd
+        values.data['mvd'] = view_definitions
+
+        
+        # comments is a free textfield, but constrainted to 256 characters
+        if len(parsed_description.comments) > 256:
+            values.data.get('validation_errors').append(values.field_name)
+            return v if type(v) == tuple else v
+
+        
+        # AddonViews cannot be used without CoordinationView.
+        view_definitions_lowercase = {mvd.lower() for mvd in view_definitions}
+        if any('addonview' in mvd for mvd in view_definitions_lowercase) and 'coordinationview' not in view_definitions_lowercase:
+            values.data.get('validation_errors').append(values.field_name)
+            return v if type(v) == tuple else v
+
+            
+        # anything besides exactly CoordinationView as a single item gets excluded
+        if len(view_definitions) == 1 and view_definitions[0].lower() != 'coordinationview':
+            values.data.get('validation_errors').append(values.field_name)
+            return v if type(v) == tuple else v
+
+        
+        return v if type(v) == tuple else v
+
+    @field_validator('mvd', mode='after')
+    def validate_and_set_mvd(cls, v, values):
+        """
+        This function runs after the other fields. It validates the mvd based on the grammar done in  
+        the 'description' field. The function checks the constraints on the mvds and returns 
+        the comma separated list into a single string.
+        """
+        view_definitions = values.data.get('mvd')
+        if not view_definitions:
+            values.data.get('validation_errors').append(values.field_name)
+            return ', '.join(view_definitions)
+
+        
+        if view_definitions == 'Not defined':
+            values.data.get('validation_errors').append(values.field_name)
+
+        
+        if values.data.get('file').schema_identifier == 'IFC4X3_ADD2':
+            view_definitions_previous_versions = {
+                'StructuralAnalysisView',
+                'SpaceBoundaryAddonView',
+                'BasicFMHandoverView',
+                'ReferenceView_V1.2',
+                'IFC4Precast'
+            }
+            if set(view_definitions) & view_definitions_previous_versions:
```

Comment:
I don't get it.

We check whether it overlaps with any of the 'previous' views, but shouldn't we just check for ReferenceView or Alignment-basedView. At least that's what the YAML previously did I suppose.

Also, why is there only a check for 4.3 add2?

Reply:
> Also, why is there only a check for 4.3 add2?

I thought that was the outcome of the discussion here https://github.com/buildingSMART/validate/pull/134#discussion_r1914846690

> In general though I don't think we can constrain 2x3 and 4 like this. The policy has always been that this is an extensible field.

Reply:
Ok, my bad then. Since these are warnings anyway, let's stick to the semantics we first had in yaml with the exception of the AddonViews.

Reply:
>  Since these are warnings anyway, let's stick to the semantics we first had in yaml with the exception of the AddonViews

[Done](https://github.com/buildingSMART/validate/pull/146/commits/59659f91adf8142ff466a8062d91c48a589f0c70)